packages/react/navigation-menu/src/NavigationMenu.tsx

```diff
@@ -496,14 +496,23 @@ interface NavigationMenuLinkProps extends PrimitiveLinkProps {
 const NavigationMenuLink = React.forwardRef<NavigationMenuLinkElement, NavigationMenuLinkProps>(
   (props: ScopedProps<NavigationMenuLinkProps>, forwardedRef) => {
     const { __scopeNavigationMenu, active, ...linkProps } = props;
+    const ref = React.useRef<NavigationMenuLinkElement>(null);
+    const composedRefs = useComposedRefs(ref, forwardedRef);
 
     return (
       <FocusGroupItem asChild>
         <Primitive.a
           data-active={active ? '' : undefined}
           aria-current={active ? 'page' : undefined}
           {...linkProps}
-          ref={forwardedRef}
+          ref={composedRefs}
+          onClick={composeEventHandlers(props.onClick, () => {
+            const rootContentDismissEvent = new Event(ROOT_CONTENT_DISMISS, {
+              bubbles: true,
+              cancelable: true,
+            });
+            ref.current?.dispatchEvent(rootContentDismissEvent);
```

Comment:
Could we do `event.target.dispatchEvent` to avoid having to setup the extra ref?